app/scripts/controllers/transactions/pending-tx-tracker.js

```
@@ -89,9 +99,14 @@ export default class PendingTransactionTracker extends EventEmitter {
   async checkUnknownTx(txMeta) {
     const txHash = txMeta.hash;
     const txId = txMeta.id;
-    const {
-      metamaskNetworkId: { name: network }
-    } = txMeta;
+    
+    // Handle both object and string formats for metamaskNetworkId
+    const metamaskNetworkId = txMeta.metamaskNetworkId;
```

ESLint

[error] Use object destructuring.

(prefer-destructuring)